### ISSUE TITLE
Error in export traces using geth

### ISSUE THREAD

**Issue by cxd1995** [blockchain-etl/ethereum-etl#140]
We have used parity to export traces. But we found the table is probably incorrect.  The data is not consistent with what we see in Etherscan. Then we try to export traces with geth.
However, when we executed the following commond: 
**ethereumetl export_traces --start-block 6000000 --end-block 6010000 \--provider-uri file://$HOME/Library/Ethereum/geth.ipc --batch-size 100 --output geth_traces.json**
We received the error below:
**ValueError: result is None in response {'jsonrpc': '2.0', 'id': 6000000, 'error': {'code': -32000, 'message': 'required historical state unavailable'}}.**
We set different number of start block and end block. BUT It did not work.
ASK FOR HELP!

**Comment by medvedev1088** [comment]
> But we found the table is probably incorrect. The data is not consistent with what we see in Etherscan.

What are the inconsistencies?

For geth there is a separate command https://github.com/blockchain-etl/ethereum-etl#export_geth_traces. You need to start geth with `geth --gcmode archive --syncmode full --ipcapi debug`.

**Comment by cxd1995** [comment]
For example, we get the following data for block 7022757 using parity:
![image](https://user-images.githubusercontent.com/45622921/50870230-25e09780-13f2-11e9-843e-821a66268d3c.png)

It dose not match with the contract internal transactions for block 7022757 on Etherscan:
![image](https://user-images.githubusercontent.com/45622921/50870140-bec2e300-13f1-11e9-9220-dd0859578dd3.png)

Particularly, the address 0x0073cf1b9230cf3ee8cab1971b8dbef21ea7b595 appearing in the trace table has no transaction or internal transaction on Ethereum at all.
![image](https://user-images.githubusercontent.com/45622921/50870956-0f880b00-13f5-11e9-9831-f2009aa40560.png)

**Comment by cxd1995** [comment]
You are right. We have changed the setting of parity
**chain = "foundation"**

**Comment by medvedev1088** [comment]
Note that syncing full archival node takes more than a month and requires more than 2TB of SSD disk space.

**Comment by cxd1995** [comment]
Thanks for your help! 
BUT there is still some problem.

Surprisingly we got sample data of trace table from block#1000000 to block#1010000 using parity. 
![image](https://user-images.githubusercontent.com/45622921/50966447-e27c4b00-150f-11e9-9c7e-d8c4a7490e24.png)
BUT there are none internal transactions for block 1000002 on Etherscan
![image](https://user-images.githubusercontent.com/45622921/50966650-98479980-1510-11e9-88df-5fcaaa236aea.png)
The data in the trace table is actually normal transactions.
![image](https://user-images.githubusercontent.com/45622921/50966637-8960e700-1510-11e9-9ecb-8f5e53c62ca1.png)

I guess there are hardly internal transactions within the first two million blocks. And why we just got normal transactions in trace table?

**Comment by medvedev1088** [comment]
That's an expected behavior. Every transaction has at least 1 trace - the top level one.

**Comment by cxd1995** [comment]
Thanks!
I tried to fiflter them out by setting trace_address is not null.
It seems works for some blocks.
![image](https://user-images.githubusercontent.com/45622921/50972612-0c3e6d80-1522-11e9-9871-8e85d2c5746b.png)

BUT when I check the data on Etherscan,  still no interal transaction were found.
![image](https://user-images.githubusercontent.com/45622921/50973268-71469300-1523-11e9-9ecc-65ad2f6799d0.png)
![image](https://user-images.githubusercontent.com/45622921/50973295-7c99be80-1523-11e9-997f-3e4d73abcc03.png)

I tried different settings of token_address. AND I found it matches perferctly with Etherscan by setting 
**trace_address = 0**
is it right?
